src/html.cr

```diff
@@ -40,9 +49,45 @@ module HTML
   # HTML.escape("Crystal & You", io) # => nil
   # io.to_s                          # => "Crystal &amp; You"
   # ```
-  def self.escape(string : String, io : IO)
+  def self.escape(string : String, io : IO, escape_quotes : Bool = true) : Nil
+    subst = ESCAPE_SUBST[escape_quotes]
+    string.each_char do |char|
+      io << subst.fetch(char, char)
+    end
+  end
+
+  # Encodes a string with JavaScript escaping substitutions.
+  #
+  # ```
+  # require "html"
+  #
+  # HTML.escape_javascript("</crystal> \u2028") # => "<\\/crystal> &#x2028;"
+  # ```
+  def self.escape_javascript(string : String) : String
+    string.gsub("\r\n", "\n").gsub(ESCAPE_JAVASCRIPT_SUBST).gsub("</", "<\\/")
+  end
+
+  # Encodes a string with JavaScript escaping, but writes to the `IO` instance provided.
+  #
+  # ```
+  # io = IO::Memory.new
+  # HTML.escape_javascript("</crystal> \u2028", io) # => nil
+  # io.to_s                                         # => "<\\/crystal> &#x2028;"
+  # ```
+  def self.escape_javascript(string : String, io : IO) : Nil
+    previous_char = '\0'
     string.each_char do |char|
-      io << SUBSTITUTIONS.fetch(char, char)
+      if previous_char == '\r' && char == '\n'
```

Comment:
Maybe this would be better as a `case` statement to express control flow more clearly and be shorter overall.
```crystal
case
when previous_char == '\r' && char == '\n'
when previous_char == '<' && char == '/'
  io << '\\' << '/'
else
  io << ESCAPE_JAVASCRIPT_SUBST.fetch(char, char)
end
previous_char = char
```